docs/modules/integrate/pages/vertx-step-by-step-buide.adoc

```diff
@@ -0,0 +1,477 @@
+        new ServerSocket(port).close();
+        return port;
+      } catch (IOException e) {
+        // port not available, try next
+      }
+    }
+    throw new RuntimeException("Could not find an available port");
+  }
+----
+
+and use it in the `start` method:
+
+[source,java]
+----
+    ...
+    int port = findFreePort(8888);
+
+    // Create the HTTP server
+    vertx.createHttpServer()
+      // Handle every request using the router
+      .requestHandler(router)
+      // Start listening
+      .listen(port)
+    ...
+----
+
+Now, we can start two instances:
+
+[source,bash]
+----
+$ java -jar target/vertx-hz-1.0.0-SNAPSHOT-fat.jar
+HTTP server started on port 8888
+Aug 30, 2024 9:09:44 AM io.vertx.launcher.application.VertxApplication
+INFO: Succeeded in deploying verticle
+
+...
+
+$ java -jar target/vertx-hz-1.0.0-SNAPSHOT-fat.jar
+HTTP server started on port 8889
+Aug 30, 2024 9:09:47 AM io.vertx.launcher.application.VertxApplication
+INFO: Succeeded in deploying verticle
+----
+
+and we can see the session is not shared between the instances, here is the request to the first instance:
+
+[source, bash]
+----
+$ http PUT localhost:8888 message="Hello world"
+HTTP/1.1 200 OK
+content-length: 28
+content-type: application/json
+set-cookie: vertx-web.session=00f219c166ca50727d23eaaf9fe54229; Path=/
+
+{
+    "messages": [
+        "Hello world"
+    ]
+}
+----
+
+and here is the request to the 2nd instance, notice the different port and that we use the cookie we received, but the data does not contain the previous message.
+
+[source, bash]
+----
+$ http PUT localhost:8889 message="Hello world 2" 'Cookie: vertx-web.session=00f219c166ca50727d23eaaf9fe54229'
+HTTP/1.1 200 OK
+content-length: 30
+content-type: application/json
+set-cookie: vertx-web.session=a1486c5ed6416972fdc356e4d91d2397; Path=/
+
+{
+    "messages": [
+        "Hello world 2"
+    ]
+}
+----
+
+We will fix that by using a Hazelcast Cluster Manager. There are two modules that provide Hazelcast Cluster Manager:
+
+- `io.vertx:vertx-hazelcast` - this module is maintained by the Vert.x team, with contributions from Hazelcast, and is built on top of open-source Hazelcast
+- `com.hazelcast:vertx-hazelcast-enterprise` / this module is maintained by the Hazelcast team and is built on top of the `vertx-hazelcast` but uses Hazelcast Enterprise instead. You need an enterprise license to use Hazelcast Enterprise.
+
+You can use either module for most of this guilde, at the end of this guide you will need the `vertx-hazelcast-enterprise` module.
+
+NOTE: You can get your trial key at https://hazelcast.com/get-started/ or you can use `vertx-hazelcast` and a community edition of Hazelcast.
+
+Add the following dependency to the `pom.xml`
+
+[source,xml]
+----
+<dependency>
+  <groupId>com.hazelcast</groupId>
+  <artifactId>vertx-hazelcast-enterprise</artifactId>
+  <version>{vertx.version}</version>
+</dependency>
+----
+
+change the following part of the `start` method
```

Comment:
where possible, add ":" (a colon) at the end of lines where information follows

Reply:
62101ff5